src/processing.py

```diff
@@ -58,8 +58,54 @@ def keep_valid_estados(df: pd.DataFrame()) -> pd.DataFrame():
     return df
 
 
+def find_next_estado(row, days_till_next_visit):
+    """
+    Keeps iterating through the observations until it finds if there are any non null ones for the days_till_next_visit time frame.
+    """
+    try:
+        days = (row["next_date_1"] - row["fecha"]).days
+        i = 1
+        while days < days_till_next_visit - 1:
+            i += 1
+            days += (row[f"next_date_{i}"] - row[f"next_date_{i-1}"]).days
+        if days <= days_till_next_visit + 1:
+            return row[f"next_estado_{i}"]
+        else:
+            return np.nan
+    except:
+        raise Exception("Next estado not found")
+
+
+def obtain_next_estados(df: pd.DataFrame, days_till_next_visit: int) -> pd.DataFrame:
+    """
+    Returns a dataset with the new y.
+    """
+    # Creating a column to display the current growth stage
+    df["estado_actual"] = df[estados].apply(find_estado_with_value_two, axis=1)
+
+    for i in range(1, days_till_next_visit):
+        df[f"next_date_{i}"] = df.groupby("codparcela")["fecha"].shift(-i)
```

Comment:
I think this can be greatly simplified.

Let's assume `days_till_next_sample=14` and `spam=1`, meaning that we are willing to accept samples between 13 and 15 samples in the future.

I think we can do:
```
Df= df.set_index(fecha).sort_index()
for i in range(days_till_next_sample - spam, days_till_next_sample + spam):
   Df[target_estado_i] = df.groupby(parcela)[estado].shift(-i, freq='D')
```

After that we should have 3 columns in this case with potential targets that would be null if there is no sample at fecha + N days. We just need to pick one from here and discard the row if all null.

We should log how many we are missing, of course.